js/service/ShareResource.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Nextcloud - News
+ *
+ * @author Marco Nassabain <marco.nassabain@hotmail.com>
+ */
+app.factory('ShareResource', function (Resource, $http, BASE_URL) {
+    'use strict';
+
+    var ShareResource = function ($http, BASE_URL) {
+        Resource.call(this, $http, BASE_URL);
+    };
+
+    ShareResource.prototype = Object.create(Resource.prototype);
+
+    ShareResource.prototype.getUsers = function (search) {
+        return this.http({
+            url: OC.linkToOCS(`apps/files_sharing/api/v1/sharees?search=${search}&itemType=file&perPage=5`, 1),
```

Comment:
What if there are users on the server that don't have access to news, only to files. They would never be able to view what you shared.

Reply:
That's true, I didn't really think about that..
We use the ocs sharee api, and i don't really see a solution from the [documentation](https://docs.nextcloud.com/server/latest/developer_manual/client_apis/OCS/ocs-sharee-api.html)

Reply:
Maybe by writing a function specific to news which does something like
```sql
select distinct(user_id) from feeds where upper(user_id) like upper(‘%query%’)
```
?

Reply:
Hmm, if that is the documented way I'm fine with it. The item type is wrong in that case though.

Reply:
There's no other itemType than "file" and "folder". If the type is different, it falls in a else and does a default behavior. 
However, I changed it to news_item it's much cleaner.